Mathlib/ModelTheory/DirectLimit.lean

```diff
@@ -59,32 +59,41 @@ variable {G' : ℕ → Type w} [∀ i, L.Structure (G' i)] (f' : ∀ n : ℕ, G'
 /-- Given a chain of embeddings of structures indexed by `ℕ`, defines a `directed_system` by
 composing them. -/
 def natLeRec (m n : ℕ) (h : m ≤ n) : G' m ↪[L] G' n :=
-  Nat.leRecOn h (fun k g => (f' k).comp g) (Embedding.refl L _)
+  Nat.leRecOn h (fun g => (f' _).comp g) (Embedding.refl L _)
 #align first_order.language.directed_system.nat_le_rec FirstOrder.Language.DirectedSystem.natLeRec
 
 @[simp]
 theorem coe_natLeRec (m n : ℕ) (h : m ≤ n) :
-    (natLeRec f' m n h : G' m → G' n) = Nat.leRecOn h fun n => f' n := by
+    (natLeRec f' m n h : G' m → G' n) = Nat.leRecOn h (f' _) := by
   obtain ⟨k, rfl⟩ := Nat.exists_eq_add_of_le h
   ext x
   induction' k with k ih
-  · rw [nat_le_rec, Nat.leRecOn_self, embedding.refl_apply, Nat.leRecOn_self]
-  ·
-    rw [Nat.leRecOn_succ le_self_add, nat_le_rec, Nat.leRecOn_succ le_self_add, ← nat_le_rec,
-      embedding.comp_apply, ih]
+  · rw [natLeRec, Nat.leRecOn_self, Embedding.refl_apply, Nat.leRecOn_self]
+  · rw [Nat.leRecOn_succ le_self_add, natLeRec, Nat.leRecOn_succ le_self_add, ← natLeRec,
+      Embedding.comp_apply, ih]
 #align first_order.language.directed_system.coe_nat_le_rec FirstOrder.Language.DirectedSystem.coe_natLeRec
 
 instance natLeRec.directedSystem : DirectedSystem G' fun i j h => natLeRec f' i j h :=
-  ⟨fun i x h => congr (congr rfl (Nat.leRecOn_self _)) rfl, fun i j k ij jk => by
-    simp [Nat.leRecOn_trans ij jk]⟩
+  ⟨fun i x h => congr (congr rfl (Nat.leRecOn_self _)) rfl, fun hij hjk => by
+    simp [Nat.leRecOn_trans hij hjk]⟩
 #align first_order.language.directed_system.nat_le_rec.directed_system FirstOrder.Language.DirectedSystem.natLeRec.directedSystem
 
 end DirectedSystem
 
 namespace DirectLimit
 
+-- Porting note : Instead of `Σi, G i`, we use the alias SigmaG which depends on `f`. This way,
+-- Lean can infer what `L` is in the `setoid` instance. Otherwise we have a
+-- "cannot find synthesization order" error. See the discussion at https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/local.20instance.20cannot.20find.20synthesization.20order.20in.20porting
+
+/-- Alias for `Σi, G i`. -/
+@[nolint unusedArguments]
+def SigmaG (_f : ∀ i j, i ≤ j → G i ↪[L] G j) := Σi, G i
```

Comment:
Just thinking out loud: should this be called something like `SigmaStr`? I'm not a big fan of names that hardcode some piece of notation, `G` is a variable after all.

How about `Language.Structure.Sigma`? Then you can write `G.Sigma`.

Also, it seems that the `Setoid` instance below doesn't just depend on `L`, but in fact depends on `f`. Which is a stronger reason to include `f` in the type. Please mention that in the porting note above. (In fact, please add that explanation to the docstring of this definition.